### PR TITLE
General code review

### PR DESCRIPTION
**Pull request by crusaderky**
- to_mongo now optionally returns a delayed instead of internally calling compute, coherently with other output functions
- general code cleanup

### CODE REVIEW

File: dask_mongo/core.py [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
+    database: str,
+    collection: str,
+) -> None:
     with pymongo.MongoClient(**connection_args) as mongo_client:
-        database_ = mongo_client.get_database(database)
-        # NOTE: `insert_many` will mutate its input by inserting a "_id" entry.
-        # This can lead to confusing results, so we make a deepcopy to avoid this.
-        database_[collection].insert_many(deepcopy(values))
+        coll = mongo_client[database][collection]
+        # insert_many` will mutate its input by inserting a "_id" entry.
```
Comment (jrbourbeau) [inline dask_mongo/core.py:21]:
```suggestion
        # `insert_many` will mutate its input by inserting a "_id" entry.
```

File: dask_mongo/core.py [hunk 2]
```diff
@@ -44,41 +43,27 @@ def to_mongo(
...
     """
-    if compute_options is None:
-        compute_options = {}
-
-    partitions = [
-        write_mongo(partition, connection_args, database, collection)
-        for partition in bag.to_delayed()
-    ]
+    partials = bag.map_partitions(write_mongo, connection_args, database, collection)
+    return checkpoint(partials)
```
Comment (jrbourbeau) [inline dask_mongo/core.py:51]: Can I ask you to add a `compute=` keyword argument (which is `True` by default) and call `.compute()` accordingly in this function? That way `to_mongo` will be consistent with other `to_parquet`, `to_csv`, `to_avro` methods in Dask.

File: dask_mongo/core.py [hunk 3]
```diff
@@ -146,17 +131,16 @@ def read_mongo(
             )
         )
 
-    partitions = [
-        fetch_mongo(
-            connection_args,
-            database,
-            collection,
+    common_args = (connection_args, database, collection, match)
```
Comment (jrbourbeau) [inline dask_mongo/core.py:134]: As is, I think we can have the same Dask collection name even when users specify different values for `chunksize`

```suggestion
    common_args = (connection_args, database, collection, chunksize, match)
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @crusaderky

File: dask_mongo/core.py (hunk 2 above)
Comment (crusaderky) [reply dask_mongo/core.py]: done

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**